pkg/didcomm/dispatcher/outbound.go

```diff
@@ -15,29 +15,55 @@ import (
 	commontransport "github.com/hyperledger/aries-framework-go/pkg/didcomm/common/transport"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/decorator"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
 )
 
 // provider interface for outbound ctx
 type provider interface {
 	Packager() commontransport.Packager
 	OutboundTransports() []transport.OutboundTransport
 	TransportReturnRoute() string
+	VDRIRegistry() vdri.Registry
 }
 
 // OutboundDispatcher dispatch msgs to destination
 type OutboundDispatcher struct {
 	outboundTransports   []transport.OutboundTransport
 	packager             commontransport.Packager
 	transportReturnRoute string
+	vdRegistry           vdri.Registry
 }
 
+const (
+	ed25519KeyType     = "Ed25519VerificationKey2018"
+	didCommServiceType = "did-communication"
+)
+
 // NewOutbound return new dispatcher outbound instance
 func NewOutbound(prov provider) *OutboundDispatcher {
 	return &OutboundDispatcher{
 		outboundTransports:   prov.OutboundTransports(),
 		packager:             prov.Packager(),
 		transportReturnRoute: prov.TransportReturnRoute(),
+		vdRegistry:           prov.VDRIRegistry(),
+	}
+}
+
+// SendToDID sends a message from myDID to the agent who owns theirDID
+func (o *OutboundDispatcher) SendToDID(msg interface{}, myDID, theirDID string) error {
+	dest, err := service.GetDestination(theirDID, didCommServiceType, ed25519KeyType, o.vdRegistry)
+	if err != nil {
+		return err
 	}
+
+	src, err := service.GetDestination(myDID, didCommServiceType, ed25519KeyType, o.vdRegistry)
+	if err != nil {
+		return err
+	}
+
+	key := src.RecipientKeys[0]
```

Comment:
why index 0 could use a comment.

Reply:
Commented, it's part of the same hardcoded key type issue.